### PR TITLE
update ypp suspension status

### PR DESCRIPTION
**Pull request by zeeshanakram3**
implements #34

### CODE REVIEW

File: packages/api/src/channels/channels.controller.ts [hunk 1]
```diff
@@ -65,16 +86,52 @@ export class ChannelsController {
...
+  }
+
+  @Put('/suspend')
+  @ApiBody({ type: SuspendChannelDto, isArray: true })
+  @ApiResponse({ type: ChannelDto, isArray: true })
+  @ApiOperation({
+    description: `Authenticated endpoint to suspend given channel/s from YPP program`,
+  })
+  async suspendChannels(
+    @Headers('ypp_owner_key') yppOwnerKey: string,
```
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts:112]: Sure.

File: packages/api/src/channels/channels.controller.ts [hunk 2]
```diff
@@ -65,16 +86,52 @@ export class ChannelsController {
...
+    try {
+      for (const { joystreamChannelId, isSuspended } of channels) {
+        const channel = await this.channelsService.get(joystreamChannelId)
+
+        // if channel is being suspended then its YT ingestion/syncing should also be stopped
+        if (isSuspended) {
+          this.channelsService.update({ ...channel, isSuspended, shouldBeIngested: false })
+        } else {
+          // if channel suspension is revoked then its YT ingestion/syncing should be resumed
+          this.channelsService.update({ ...channel, isSuspended, shouldBeIngested: true })
```
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts:128]: Right, so I think irrespective of the previous YT sync status, whether enabled or disabled, if a channel gets unsuspended, its YT sync should always be in the disabled state. And it should be the responsibility of the channel owner to turn it back on?

File: packages/api/src/channels/channels.controller.ts [hunk 3]
```diff
@@ -65,16 +86,52 @@ export class ChannelsController {
     }
   }
 
-  @Put(':joystreamChannelId')
+  @Put('/yt-syncing/:joystreamChannelId')
```
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts:89]: Right, I also initially thought of keeping the endpoint like `/channels/:channelId`. However, we actually need two different endpoints for changing two different fields/values (`shouldBeIngested` & `isSuspended`) of the same resource (which is the channel record) by different actors, having different privileges. 

So maybe two endpoints like the following make sense?

- `/channels/ingest/:channelId` -> callable by channel owners for changing channel's ingestion status, would change `shouldBeIngested` field 
- `/channels/suspend/:channelId` -> callable by YPP infra owner to suspend channel's YPP status, would change `isSuspended` field

File: packages/api/src/channels/channels.controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kdembler) [reply packages/api/src/channels/channels.controller.ts:128]: Sounds good

File: packages/api/src/channels/channels.controller.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kdembler) [reply packages/api/src/channels/channels.controller.ts:89]: Sure sounds good but I would change the order to `/channels/:channelId/ingest` and same for suspend

File: packages/api/src/channels/channels.controller.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts]: Addressed in 2aad464

File: packages/api/src/channels/channels.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts]: Addressed in 2aad464

File: packages/api/src/channels/channels.controller.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (zeeshanakram3) [reply packages/api/src/channels/channels.controller.ts]: Addressed in 2aad464

**Pull request merged**